backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/RemoveImageCommand.java

```diff
@@ -323,4 +331,24 @@ protected VDSReturnValue performDeleteImageVdsmOperation() {
                                 getDiskImage().isWipeAfterDelete(), getStorageDomain().getDiscardAfterDelete(),
                                 getParameters().isForceDelete())));
     }
+
+    @Override
+    public CommandCallback getCallback() {
+        return getParameters().getParentCommand() == ActionType.RemoveDisk ?
+                callbackProvider.get() :
+                null;
```

Comment:
what happens when the callback is null - will onSucceeded be called after successful execution of the `execute` method?

Reply:
Good point